### PR TITLE
Fixing bug in url query reader and added test cases

### PR DESCRIPTION
**Pull request by tariq1890**
Fixing a bug where url query strings are parsed incorrectly.

url.ParseQuery expects a query string to be passed rather than the entire URL. In the current code, one would never be able to retrieve the actual value of the first query key of the URL. This PR will fix that.

Also added a few test cases

### CODE REVIEW

**Comment by marefr** [comment]
Thanks. Would be good if you could open an issue to track this where you describe the bug, what was expected and what happened instead.

**Comment by tariq1890** [comment]
@marefr  I found this bug while trying to write test cases for that method. I have changed the code so that it passes the test cases that I have written. (Those tests will fail with the current code). 

Please let me know if my test cases are valid. Also, If you'd like me to reproduce this, I can work towards setting it up locally and running grafana to reproduce it.

**Comment by tariq1890** [comment]
So I have pointed out an issue in the usage of the url.ParseQuery (https://github.com/grafana/grafana/pull/12815/files#diff-68303a2aea932f2982a2fae85d4bf897R13)where you are passing an entire URL string when one is actually supposed to pass in only the query string. The godoc of url.ParseQuery mentions the very same thing.

Like I said before, I can definitely work towards reproducing it. I feel the unit tests here have helped uncover this potential bug. I've added enough test cases to cover and demonstrate that the change is safe.

**Comment by marefr** [comment]
My point being the only code using that method is https://github.com/grafana/grafana/blob/master/pkg/api/render.go#L17 and we haven't received any bug reports for that. But you provide a good pointer and we should probably fix that.

File: pkg/util/url_test.go [hunk 1]
```diff
@@ -43,4 +44,36 @@ func TestUrl(t *testing.T) {
...
 	})
+
+	Convey("When joining two urls where lefthand side has a trailing slash and righthand side has preceding slash", t, func() {
+		result := JoinUrlFragments("http://localhost:8080/", "/api/")
+
+		So(result, ShouldEqual, "http://localhost:8080/api/")
+	})
+}
+
+func TestNewUrlQueryReader(t *testing.T) {
```
Comment (marefr) [inline pkg/util/url_test.go]: You're not using goconvey - please fix and/or merge with url test above with multiple    
subsequent Convey statements.

**Review by marefr — CHANGES_REQUESTED** [review]
Some minor changes needed.

Validation tests are needed and appreciated

**Comment by tariq1890** [comment]
Done. Thanks for the review. This bug would not have surfaced as the default parameter is always there to be returned as backup. I reckon this issue would have been noticed only if a person is looking for a particular width of the PNG and that width wasn't used during render.

**Review by marefr — APPROVED** [review]

**Pull request merged**